File: Kotlin/InsertionSort.kt
fun main(args: Array<String>) {
    //Input the array in the format num1 num2 num3 num4 etc
    print("Enter the elements for the Array: ")
    var inputArray = readLine()!!.trim().split(" ").map { it -> it.toInt() }.toIntArray()

    var sortedArray = insertionSort(inputArray)
    println(sortedArray.contentToString())
}

fun insertionSort(arr: IntArray): IntArray {
    for (i in 0 until arr.lastIndex) {
        for (j in i+1 downTo 1) {
            if(arr[j]<arr[j-1]) {
                var temp = arr[j - 1]
                arr[j - 1] = arr[j]
                arr[j] = temp
            }
            else
                break
        }
    }
    return arr
}


//Sample
//Input:   Enter the elements for the Array: 5 0 -23 -2 15 1 62
//Output:   [-23, -2, 0, 1, 5, 15, 62]

//Time Complexity:  Worst Case: O(n^2)   Best Case: O(n)
//Space Complexity: O(1)

File: Python/Insertion_Sort.py
'''
Insertion Sort implementation in Python
Input: Array that user want to sort
Output: Sorted array in ascending order

----------------------------------------

Sample Input 1:
6
[6, 4, 2, 1, 78, 23]
Output:
Sorted Array: 1 2 4 6 23 78
Sample Input 2:
5
[34, 56, 78, 98, 99]
Output:
Sorted Array: 34 56 78 98 99

----------------------------------------

Time Complexity:
Worst and Average case = O(n^2)
Best case (when array is already sorted) = O(n)
Space Complexity: O(1)
'''

l=list(map(int,input().split()))    #take array from user
for i in range(1,len(l)):
    j=i-1
    key=l[i]
    while j>=0 and key<l[j]:
        l[j+1]=l[j]
        j-=1
    l[j+1]=key
print(l)    # print sorted array